src/asgen/backends/alpinelinux/apkpkgindex.d

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (C) 2020 Rasmus Thomsen <oss@cogitri.dev>
+ *
+ * Licensed under the GNU Lesser General Public License Version 3
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the license, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+module asgen.backends.alpinelinux.apkpkgindex;
+
+import std.algorithm : canFind, filter, endsWith;
+import std.array : appender, split;
+import std.conv : to;
+import std.exception : enforce;
+import std.file : dirEntries, exists, SpanMode;
+import std.format : format;
+import std.path : baseName, buildPath;
+import std.string : splitLines, startsWith, strip;
+import std.utf : UTFException, validate;
+
+import asgen.logging;
+import asgen.zarchive;
+import asgen.utils : escapeXml;
+import asgen.backends.interfaces;
+import asgen.backends.alpinelinux.apkpkg;
+
+final class AlpinePackageIndex : PackageIndex
+{
+
+private:
+    string rootDir;
+    Package[][string] pkgCache;
+
+public:
+
+    this(string dir)
+    {
+        enforce(exists(dir), format("Directory '%s' does not exist.", dir));
+        this.rootDir = dir;
+    }
+
+    override void release()
+    {
+        pkgCache = null;
+    }
+
+    private void setPkgDescription(AlpinePackage pkg, string pkgDesc)
+    {
+        if (pkgDesc is null)
+        {
+            return;
```

Comment:
style: Skip braces for simple one-line statements after a conditional (especially if it's a return statement)